src/components/share-button/share-button.component.ts

```diff
@@ -37,64 +35,34 @@ export class ShareButtonComponent implements AfterViewInit {
     @Input() count: boolean = false;
     /** Output button count to calculate total share counts */
     @Output() countOuter = new EventEmitter<number>();
-
     /** Output pop up closed*/
     @Output() popUpClosed = new EventEmitter<ShareProvider>();
 
-    @ViewChild('btn') btn: ElementRef;
-
-    private window: Window;
+    window: Window;
+    self: HTMLElement;
+    btn: HTMLElement;
 
     constructor(private sbService: ShareButtonsService,
-        private renderer: Renderer,
-        private elementRef: ElementRef,
-        window: WindowService) {
+                private renderer: Renderer,
+                elementRef: ElementRef,
+                window: WindowService) {
         this.window = window.nativeWindow;
+        this.self = elementRef.nativeElement;
     }
 
     ngAfterViewInit() {
-        /** If URL is not presented then set the current URL    */
-        if (this.url) {
-            /** If URL is presented check if it is a valid URL */
-            let r = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/;
-            if (!r.test(this.url)) {
-                console.warn('ShareButtons: Invalid URL, switching to window.location.href');
-                /** Use encodeURIComponent to get the full URL including after the hash */
-                this.url = this.window ?
-                    encodeURIComponent(this.window.location.href)
-                    : typeof global !== 'undefined' ? (<any>global).url : '';
-            }
-        } else {
-            /** This supposed to fix window when undefined on Universal */
-            this.url = this.window ?
-                encodeURIComponent(this.window.location.href) :
-                typeof global !== 'undefined' ? (<any>global).url : '';
-        }
-
-        /** Set button template */
-        this.renderer.setElementProperty(this.btn.nativeElement, 'innerHTML', this.button.template);
-
-        /** Set buttons classes */
-        let classes = this.button.classes.match(/\S+/g) || [];
-        classes.map((btnClass) => this.renderer.setElementClass(this.btn.nativeElement, btnClass, true));
-
+        /** Validate URL */
+        this.url = Helper.validateUrl(this.url, this.window);
```

Comment:
Moving validation logic to `Helper.validateUrl() ` is the right think to do 👍 
But instead of passing  the entire window object to `validateUrl()`, which would make the method harder to test 
I suggest keep the default location retrieval in the component and update the method as such:

`validateUrl(url: string, defaultUrl?:string)`

UI related code should be keept in components and not be move to services